### PR TITLE
#7: Step 4 (closes #7)

### LINKED ISSUE #7: Step 4
## requirements
Use `enumero` to define `Move`

## specs
- use enumero (https://github.com/buildo/enumero) to encode `Move` 
- use `indexedEnum` from enumero, using `String` as index
- use `CaseEnumSerialization` to print to string and `CaseEnumIndex` to parse from index

### Relevant points

- enumero standardizes the ADT technique we’ve seen above
- the API is still under development and suggestions are welcome :-)

### Solution
https://github.com/buildo/rps/compare/step-3...step-4

### PR DESCRIPTION
**Pull request by raasoft**
Closes #7

### tests performed

kapture 2018-07-12 at 14 58 48

### tests not performed (domain coverage)

### CODE REVIEW

File: src/main/scala/rps/Game.scala [hunk 1]
```diff
@@ -10,48 +11,37 @@ object Game {
...
     }
   }
 
-    private def generateCPUMove(): RPSMove = {
+    private def generateCPUMove(): RPSMoves = {
       import scala.util.Random
-      Random.shuffle(Set(Rock, Paper, Scissors)).head
+      /* Apparently there is no way of obtaining 
+      the set of all enumerators from a CaseEnumIndex
+      so I am going to recreate everything here.*/
```
Comment (gabro) [inline src/main/scala/rps/Game.scala:40]: yup, this is missing in the API of CaseEnumIndex apparently, sorry about that.

It's present in `CaseEnum` though, which we will use in the next steps

File: src/main/scala/rps/RPSMoves.scala [hunk 2]
```diff
@@ -9,28 +10,18 @@ a little bit too much for the average use-case.
...
-
-    def factoryMove(aMoveString: String): Option[RPSMove] = {
-        setOfMoves.filter(_.value == aMoveString).headOption
-    }
+/* 
+I will use the @indexedEnum annotation, that
+builds enumerations that follow the library's 
+convention for ADT-based enums with an associated value.
+*/
+@indexedEnum trait RPSMoves {
```
Comment (gabro) [inline src/main/scala/rps/RPSMoves.scala:22]: ultra-minor, but usually we use the singular form for the name.

**Review by gabro — APPROVED** [review]
👍 good! Sorry about the lack of `values` on `CaseEnumIndex`!

File: src/main/scala/rps/RPSMoves.scala (hunk 2 above)
Comment (raasoft) [reply src/main/scala/rps/RPSMoves.scala:22]: Will be changed in the next revision!

**Pull request merged**